drivers/pio/sunxi-pio.c

```diff
@@ -14,56 +15,49 @@
 #include <drivers/pio.h>
 #include <drivers/pio/sunxi-pio.h>
 
-/* This line goes in the header. */
-#define SUNXI_PIO_PIN(port, index) ((port) << 5 | ((index) & 0x1f))
+#define MAX_PORTS         8
+#define MAX_PINS_PER_PORT 32
 
-#define MAX_PORTS                  8
-#define MAX_PINS_PER_PORT          32
+#define PIN_INDEX(pin)    ((pin) & BITMASK(0, 5))
+#define PIN_PORT(pin)     ((pin) >> 5)
 
-#define PIN_INDEX(pin)             ((pin) & BITMASK(0, 5))
-#define PIN_PORT(pin)              ((pin) >> 5)
-
-#define PIN_MODE(mode)             ((mode) & BITMASK(0, 3))
+#define PIN_MODE(mode)    ((mode) & BITMASK(0, 3))
 
 #define CONFIG_REG(pin) \
```

Comment:
As written, all of these macros require a pin number -- starting counting from the beginning of the device (port 0 pin 0) -- as returned by `SUNXI_PIO_PIN` above (which, as noted, should be moved to the header). They don't work with just a pin offset from the beginning of a port.

Reply:
This has been added to the header.